backend/dataall/modules/datapipelines/services/datapipelines_permissions.py

```diff
@@ -0,0 +1,50 @@
+from dataall.db.permissions import ENVIRONMENT_INVITED, ENVIRONMENT_INVITATION_REQUEST, \
+    ENVIRONMENT_ALL, TENANT_ALL, TENANT_ALL_WITH_DESC, RESOURCES_ALL, RESOURCES_ALL_WITH_DESC
+
+
+"""
+DATAPIPELINE PERMISSIONS FOR ENVIRONMENT
+"""
+CREATE_PIPELINE = 'CREATE_PIPELINE'
+LIST_PIPELINES = 'LIST_PIPELINES'
+
+ENVIRONMENT_INVITED.append(CREATE_PIPELINE)
+ENVIRONMENT_INVITED.append(LIST_PIPELINES)
+
+ENVIRONMENT_INVITATION_REQUEST.append(CREATE_PIPELINE)
+
+ENVIRONMENT_ALL.append(CREATE_PIPELINE)
+ENVIRONMENT_ALL.append(LIST_PIPELINES)
+
+RESOURCES_ALL.append(CREATE_PIPELINE)
+RESOURCES_ALL.append(LIST_PIPELINES)
+
+RESOURCES_ALL_WITH_DESC[CREATE_PIPELINE] = CREATE_PIPELINE
+RESOURCES_ALL_WITH_DESC[LIST_PIPELINES] = LIST_PIPELINES
+
+
+"""
+DATAPIPELINES
+"""
+MANAGE_PIPELINES = 'MANAGE_PIPELINES'
+
+TENANT_ALL.append(MANAGE_PIPELINES)
+TENANT_ALL_WITH_DESC[MANAGE_PIPELINES] = 'Manage pipelines'
+
+GET_PIPELINE = 'GET_PIPELINE'
+UPDATE_PIPELINE = 'UPDATE_PIPELINE'
+DELETE_PIPELINE = 'DELETE_PIPELINE'
+CREDENTIALS_PIPELINE = 'CREDENTIALS_PIPELINE'
+PIPELINE_ALL = [
+    CREATE_PIPELINE,
+    GET_PIPELINE,
+    UPDATE_PIPELINE,
+    DELETE_PIPELINE,
+    CREDENTIALS_PIPELINE,
+    LIST_PIPELINES,
+]
+
+RESOURCES_ALL.extend(PIPELINE_ALL)
+for perm in PIPELINE_ALL:
```

Comment:
Optional. Hm, looks like the code for permissions repeats, we could create an API in permissions.py to conveniently insert all needed permissions. For instance, some methods like `add_to_invite`, `add_all_to_resources` or something like that

Reply:
I agree, but I will do it in a separate followup PR, and adapt the code in all the modules to use it